src/chunktransformer.js

```diff
@@ -0,0 +1,337 @@
+  }
+  this.options = options || {};
+  this.options.objectMode = true; // forcing object mode
+  Transform.call(this, options);
+  this._destroyed = false;
+  this.reset();
+}
+util.inherits(ChunkTransformer, Transform);
+/**
+ * transform the readrowsresponse chunks into friendly format. Chunks contain 3 properties:
+ *
+ * `rowContents` The row contents, this essentially is all data pertaining
+ *     to a single family.
+ *
+ * `commitRow` This is a boolean telling us the all previous chunks for this
+ *     row are ok to consume.
+ *
+ * `resetRow` This is a boolean telling us that all the previous chunks are to
+ *     be discarded.
+ *
+ * @public
+ *
+ * @param {object} data readrows response containing array of chunks.
+ * @param {object} [enc] encoding options.
+ * @param {callback} next callback will be called once data is processed, with error if any error in processing
+ */
+ChunkTransformer.prototype._transform = function(data, enc, next) {
+  try {
+    for (const chunk of data.chunks) {
+      switch (this.state) {
+        case RowStateEnum.NEW_ROW:
+          this.processNewRow(chunk, this.options);
+          break;
+        case RowStateEnum.ROW_IN_PROGRESS:
+          this.processRowInProgress(chunk, this.options);
+          break;
+        case RowStateEnum.CELL_IN_PROGRESS:
+          this.processCellInProgress(chunk, this.options);
+          break;
+      }
+    }
+    next();
+  } catch (err) {
+    next(err);
+  }
+};
+
+/**
+ * called at end of the stream.
+ * @public
+ * @param {callback} callback callback will be called with error if there is any uncommitted row
+ */
+ChunkTransformer.prototype._flush = function(cb) {
+  try {
+    this.invariant(
+      typeof this.row.key !== 'undefined',
+      'Response ended with pending row without commit',
+      this.row
+    );
+    cb();
+  } catch (err) {
+    cb(err);
+  }
+};
+
+/**
+ * called when stream is destroyed.
+ * @public
+ * @param {error} error error if any
+ */
+ChunkTransformer.prototype.destroy = function(err) {
+  if (this._destroyed) return;
+  this._destroyed = true;
+  var self = this;
+  process.nextTick(function() {
+    if (err) {
+      self.emit('error', err);
+    }
+    self.emit('close');
+  });
+};
+
+/**
+ * Resets state of formatter
+ * @private
+ */
+ChunkTransformer.prototype.reset = function() {
+  this.prevRowKey = '';
+  this.family = {};
+  this.qualifiers = [];
+  this.qualifier = {};
+  this.row = {};
+  this.state = RowStateEnum.NEW_ROW;
+};
+/**
+ * sets prevRowkey and calls reset when row is committed.
+ * @private
+ */
+ChunkTransformer.prototype.commit = function() {
+  const row = this.row;
+  this.reset();
+  this.prevRowKey = row.key;
+};
+/**
+ * Error checker if true throws error
+ * @private
+ * @param {boolean} condition condition to evaluate
+ * @param {string} text Error text
+ * @param {object} chunk chunk object to append to text
+ */
+ChunkTransformer.prototype.invariant = function(condition, text, chunk) {
```

Comment:
When scanning the code, it's not clear that any call to `this.invariant()` could result in a thrown error. I would rather see something like this, so we aren't try/catching anywhere:

```diff
- this.invariant(condition, 'Error message.', value);
+ if (condition) {
+   this.destroy(this.generateError('Error message.', value));
+ }
```

You could use [create-error-class](https://github.com/floatdrop/create-error-class):

```js
const createErrorClass = require('create-error-class');

const TransformError = createErrorClass('TransformError', function (props) {
  this.message = `${props.message}: ${JSON.stringify(props.chunk)}`;
});

// ...

ChunkTransformer.prototype.validationMethod = function(chunk) {
  if (condition) {
    this.destroy(new TransformError({
      message: 'Error message.',
      chunk: chunk
    });
  }
};
```